charms/openfga-k8s/src/charm.py

```diff
@@ -184,20 +185,36 @@ def _on_update_status(self, _):
         self._ready()
 
     @requires_state
-    def _create_token_secret(self, event):
-        if not self._state.token_secret_id:
+    def _create_token(self, event):
+        if JujuVersion.from_environ().has_secrets:
+            if not self._state.token_secret_id:
+                token = secrets.token_urlsafe(32)
+                content = {"token": token}
+                secret = self.app.add_secret(content)
+                self._state.token_secret_id = secret.id
+                logger.info("created token secret {}".format(secret.id))
+        else:
             token = secrets.token_urlsafe(32)
-            content = {"token": token}
-            secret = self.app.add_secret(content)
-            self._state.token_secret_id = secret.id
-            logger.info("created token secret {}".format(secret.id))
+            self._state.token = token
+
+    @requires_state
+    def _get_token(self, event):
+        if JujuVersion.from_environ().has_secrets:
+            if self._state.token_secret_id:
+                secret = self.model.get_secret(id=self._state.token_secret_id)
+                secret_content = secret.get_content()
+                return secret_content["token"]
+            else:
+                return None
```

Comment:
This else block runs when the Juju installation supported secrets but we don't have a secret ID, right? If so, shouldn't we set some status message here, like awaiting secret, before returning?

Reply:
I think we can leave the status update to the caller.

Reply:
no, this else block runs when juju supports secrets, but the token_secret_id has not been set yet on the state..